### PR TITLE
[Parser] switch to source id API

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #4849

### Summary of Changes

This PR continues @yurydelendik's work to switch to a `sourceId` based API. This saves us ~100ms when we're passing large objects back and forth like a source w/ text.

screen shot 2018-01-12 at 3 05 24 pm

### CODE REVIEW

File: src/actions/breakpoints.js [hunk 1]
```diff
@@ -339,7 +339,7 @@ export function toggleBreakpoint(line: ?number, column?: number) {
         })
       );
     }
-
+    console.log(selectedSource.toJS());
```
Comment (bomsy) [inline src/actions/breakpoints.js]: we can remove this

File: src/actions/breakpoints/addBreakpoint.js [hunk 2]
```diff
@@ -50,7 +53,8 @@ export default async function addBreakpoint(
     newGeneratedLocation
   );
 
-  const astLocation = await getASTLocation(sourceRecord, newLocation);
+  console.log("meh", source);
```
Comment (bomsy) [inline src/actions/breakpoints/addBreakpoint.js]: meh :)

File: src/actions/breakpoints/addBreakpoint.js [hunk 3]
```diff
@@ -50,7 +53,8 @@ export default async function addBreakpoint(
     newGeneratedLocation
   );
 
-  const astLocation = await getASTLocation(sourceRecord, newLocation);
+  // console.log("meh", source);
```
Comment (jasonLaster) [inline src/actions/breakpoints/addBreakpoint.js]: :/

File: src/test/tests-setup.js [hunk 4]
```diff
@@ -42,7 +42,7 @@ const config = {
 };
 
 global.DebuggerConfig = config;
-
+global.performance = { now: () => {} };
```
Comment (jasonLaster) [inline src/test/tests-setup.js]: :/

**Pull request merged**